src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAzureUtilityImpl.java

```diff
@@ -0,0 +1,243 @@
+package ca.corefacility.bioinformatics.irida.repositories.filesystem;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.GZIPInputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.util.FileUtils;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.models.BlobStorageException;
+
+/**
+ * Component implementation of file utitlities for azure storage
+ */
+@Component
+public class IridaFileStorageAzureUtilityImpl implements IridaFileStorageUtility {
+	private static final Logger logger = LoggerFactory.getLogger(IridaFileStorageAzureUtilityImpl.class);
+
+	private BlobServiceClient blobServiceClient;
+	private BlobContainerClient containerClient ;
+	private BlobClient blobClient;
+
+	@Autowired
+	public IridaFileStorageAzureUtilityImpl(String connectionStr, String containerName){
+		this.blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionStr)
+				.buildClient();
+		this.containerClient = blobServiceClient.getBlobContainerClient(containerName);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public File getTemporaryFile(Path file) {
+		File fileToProcess = null;
+
+		// We set the blobClient "path" to which we want to upload our file to
+		blobClient = containerClient.getBlobClient(getAzureFileAbsolutePath(file));
+
+		try {
+			// Create a file that will be unique in the /tmp/ folder. We append the current date/time
+			// to the file name
+			String tmpDir = "/tmp/" + new Date().toString().replaceAll("\\W", "");
```

Comment:
This has been completely changed. If you take a look at the object_store/_analysis_submissions PR, I've made use of the existing folders

Reply:
I see the `getFile` method in that branch, but it looks like it might have a slightly different use?  I understand the change later in the chain, but can it get pulled down to this PR since it was found here?  Its too easy for temporary stuff like this to accidentally slip by when getting reviewed later.

Reply:
Sure I can pull it down into here. The use is the same just renamed the method in a later branch

Reply:
Updated in 23c0b6d48cdf222b4c5674c280f79ed361766082